indra/newview/bdfloaterposecreator.cpp

```diff
@@ -0,0 +1,3067 @@
+                    LLKeyframeMotion::PositionKey position_key = LLKeyframeMotion::PositionKey(1.f, joint->getTargetPosition());
+                    joint_motion->mPositionCurve.mKeys[0] = position_key;
+                    joint_state->setUsage(joint_state->getUsage() | LLJointState::POS);
+                }*/
+
+                temp_motion->addJointState(joint_state);
+
+                //BD - Start with filling general scale data in.
+                joint_motion->mScaleCurve.mNumKeys = 0;
+                joint_motion->mScaleCurve.mInterpolationType = LLKeyframeMotion::IT_LINEAR;
+
+                //BD - Create a scale key and put it into the scale curve.
+                /*LLKeyframeMotion::ScaleKey scale_key = LLKeyframeMotion::ScaleKey(1.f, joint->getScale());
+                joint_motion->mScaleCurve.mKeys[0] = scale_key;*/
+
+                joint_motion->mUsage = joint_state->getUsage();
+
+                //BD - We do not use constraints so we just leave them out here.
+                //     Should we ever add them we'd do so here.
+                //joint_motion_list->mConstraints.push_front(constraint);
+
+                //BD - Get the pelvis's bounding box and add it.
+                if (joint->getJointNum() == 0)
+                {
+                    //mTempMotionList->mPelvisBBox.addPoint(position_key.mPosition);
+                    mTempMotionList->mPelvisBBox.addPoint(joint->getTargetPosition());
+                }
+            }
+        }
+    }
+
+    temp_motion->setJointMotionList(mTempMotionList);
+    //BD - We export this file immediately so we can import it later.
+    //     This is going to be the animation we write everything we do into
+    //     and it also neatly acts as an "autosave" thing that we can just
+    //     load back in case we quit out of the pose creator.
+    //     The main reason we do this however is because we are going to import
+    //     this animation locally which will create a fully setup LLKeyframeAnimation
+    //     that doesn't have some weird things going on causing crashes, it is also
+    //     a lot easier to do.
+    std::string full_path = gDirUtilp->getExpandedFilename(LL_PATH_USER_SETTINGS, "animations", "_poser_temp.anim");
+    temp_motion->dumpToFile(full_path);
+    //return temp_motion;
+}
+
+//BD - This is used to collect all default values at the beginning to revert to later on.
+void BDFloaterPoseCreator::onCollectDefaults()
+{
+    LLQuaternion rot;
+    LLVector3 pos;
+    LLVector3 scale;
+    LLJoint* joint;
+
+    //BD - Getting collision volumes and attachment points.
+    std::vector<std::string> joint_names, cv_names, attach_names;
+    gAgentAvatarp->getSortedJointNames(0, joint_names);
+    gAgentAvatarp->getSortedJointNames(1, cv_names);
+    gAgentAvatarp->getSortedJointNames(2, attach_names);
+
+    mDefaultRotations.clear();
+    mDefaultScales.clear();
+    mDefaultPositions.clear();
+
+    for (auto name : joint_names)
```

Comment:
Why not `for (const auto& name : joint_names)`? Pretty sure I there are other cases in this file as well, but I didn't check closely.